### PR TITLE
Fix up errorprone and warnings that fail the check build

### PR DESCRIPTION
**Pull request by breedx-splk**
This is related to upstream https://github.com/open-telemetry/opentelemetry-android/pull/46 and https://github.com/open-telemetry/opentelemetry-android/pull/48.

We had a number of issues that were not caught when errorprone started not working on android builds. This should help with those.

### CODE REVIEW

File: splunk-otel-android/src/main/java/com/splunk/rum/LogToSpanBridge.java [hunk 1]
```diff
@@ -34,6 +34,7 @@
 import io.opentelemetry.sdk.logs.data.LogRecordData;
 import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
 import java.util.concurrent.TimeUnit;
+import org.jetbrains.annotations.Nullable;
```
Comment (laurit) [inline splunk-otel-android/src/main/java/com/splunk/rum/LogToSpanBridge.java:37]: VolleyHttpClientAttributesGetter uses `androidx.annotation.Nullable`, elsewhere we use `javax.annotation.Nullable`

File: opentelemetry-android-instrumentation/src/main/java/io/opentelemetry/rum/internal/instrumentation/activity/ActivityTracer.java [hunk 2]
```diff
@@ -195,13 +198,22 @@ private String getActivityName() {
...
-
         public Builder setScreenName(String screenName) {
             this.screenName = screenName;
             return this;
         }
+
+        public ActivityTracer build() {
+            if (activeSpan == INVALID_ACTIVE_SPAN) {
+                throw new IllegalStateException("activeSpan must be configured.");
+            }
```
Comment (mateuszrzeszutek) [inline opentelemetry-android-instrumentation/src/main/java/io/opentelemetry/rum/internal/instrumentation/activity/ActivityTracer.java:209]: Why not `if (activeSpan == null)`?
I'm guessing that'd require adding some `@Nullable` and `requireNonNull()`, but isn't that less bad than creating fake invalid instances/

Comment (breedx-splk) [reply opentelemetry-android-instrumentation/src/main/java/io/opentelemetry/rum/internal/instrumentation/activity/ActivityTracer.java:209]: Let's see what it looks like....

Comment (mateuszrzeszutek) [reply opentelemetry-android-instrumentation/src/main/java/io/opentelemetry/rum/internal/instrumentation/activity/ActivityTracer.java:209]: Bleh... welp, too bad, at least we tried.

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**